exchange/bitswap/bitswap.go

```diff
@@ -385,7 +404,15 @@ func (bs *Bitswap) wantNewBlocks(ctx context.Context, bkeys []u.Key) {
 			}
 		}(p)
 	}
-	wg.Wait()
+	done := make(chan struct{})
+	go func() {
+		wg.Wait()
+		close(done)
+	}()
+	select {
+	case <-done:
+	case <-ctx.Done():
+	}
```

Comment:
- https://github.com/jbenet/go-peerstream/blob/master/transport/yamux/yamux.go
- https://github.com/hashicorp/yamux
- https://godoc.org/github.com/hashicorp/yamux

cant readily see anything im doing wrong. maybe it does wait for an ack to do flowcontrol / backpressure.